### ISSUE TITLE
Non-trivial designated initializers not supported

### ISSUE THREAD

**Issue by jpetso** [couchbaselabs/cbforest#8]
The other remaining issue with my Linux build (at least as far as I made it in the build) is this piece of code (from KeyStore.cc) and a similar one two functions down:
```
        fdb_doc doc = {
            .key = (void*)key.buf,
            .keylen = key.size,
            .meta = (void*)meta.buf,
            .metalen = meta.size,
            .body  = (void*)body.buf,
            .bodylen = body.size,
        };
```

Which g++ (GCC 4.9.2 20150304 (prerelease)) doesn't quite like too much:
```
FAILED: /usr/bin/c++    -std=c++11 -fomit-frame-pointer -O2 -g -DNDEBUG -I../vendor/forestdb/include -I.. -MMD -MT CMakeFiles/cbforest.dir/CBForest/KeyStore.cc.o -MF CMakeFiles/cbforest.dir/CBForest/KeyStore.cc.o.d -o CMakeFiles/cbforest.dir/CBForest/KeyStore.cc.o -c ../CBForest/KeyStore.cc
../CBForest/KeyStore.cc: In member function ‘forestdb::sequence forestdb::KeyStoreWriter::set(forestdb::slice, forestdb::slice, forestdb::slice)’:
../CBForest/KeyStore.cc:116:9: sorry, unimplemented: non-trivial designated initializers not supported
         };
         ^
../CBForest/KeyStore.cc: In member function ‘bool forestdb::KeyStoreWriter::del(forestdb::slice)’:
../CBForest/KeyStore.cc:149:9: sorry, unimplemented: non-trivial designated initializers not supported
         };
         ^
```

The reason is that this is not valid C++, neither C++11 nor C++14, although it is valid C99. (C++ initializer lists got adopted, but designated initializers didn't.) In the interest of portability, maybe it can be swapped out with something more standard.

**Comment by snej** [comment]
Sure, could you send a pull request?

**Comment by jpetso** [comment]
Thanks, added a commit to PR #6.

**Comment by jpetso** [comment]
Fixed by commit 77c403c87ab0e6b12eae26f72edfd0179405b4ab.
